### PR TITLE
add functions for 2.0

### PR DESCRIPTION
**Pull request by Amber1990Zhang**

### CODE REVIEW

**Review by whitewum — COMMENTED** [review]
minor about NULL test

File: docs_2.0/manual_EN/4.ngql-guide/6.functions/2.string.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Built-in string functions
+
+Nebula Graph supports the following built-in string functions:
+
+> **NOTE:** Like SQL, the character index (location) for nGQL starts from `1`. However, the character index (location) for C language starts from `0`.
```
Comment (randomJoe211) [inline docs_2.0/manual_EN/4.ngql-guide/6.functions/2.string.md:5]: for the C language starts from
Here lacks a "the".

File: docs_2.0/manual_EN/4.ngql-guide/6.functions/2.string.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+Function| Description |
+----  |  ----|
+int strcasecmp(string a, string b) | Compare strings without case sensitivity, when a = b, return 0, when a > b returned value is greater than 0, otherwise less than 0 |
+string lower(string a) | Return the argument in lowercase |
+string upper(string a) | Return the argument in uppercase |
+int length(string a) | Return length (int) of given string in bytes |
```
Comment (randomJoe211) [inline docs_2.0/manual_EN/4.ngql-guide/6.functions/2.string.md:12]: Lacking the: Return the length (int) of the given string

File: docs_2.0/manual_EN/4.ngql-guide/6.functions/2.string.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+string trim(string a) | Remove leading and trailing spaces |
+string ltrim(string a) | Remove leading spaces |
+string rtrim(string a) | Remove trailing spaces |
+string left(string a, int count) | Return the substring in [1, count], if length a is less than count, return a |
+string right(string a, int count) | Return the substring in [size - count + 1, size], if length a is less than count, return a |
+string lpad(string a, int size, string letters) | Left-pads a string with another string to a certain length|
+string rpad(string a, int size, string letters)| Reft-pads a string with another string to a certain length  |
+string substr(string a, int pos, int count) | Extract a substring from a string, starting at the specified position, extract the specified length characters |
+int hash(string a) | Encode the data into integer value |
```
Comment (randomJoe211) [inline docs_2.0/manual_EN/4.ngql-guide/6.functions/2.string.md:21]: Lacking article: Encode the data into an integer value/ into integer values

File: docs_2.0/manual_EN/4.ngql-guide/6.functions/2.string.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+**Explanations** on the returns of function `substr`:
+
+- If pos is 0, return empty string
```
Comment (randomJoe211) [inline docs_2.0/manual_EN/4.ngql-guide/6.functions/2.string.md:25]: Lacking article: return an empty string

File: docs_2.0/manual_EN/4.ngql-guide/6.functions/2.string.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+- If the absolute value of pos is greater than the string, return empty string
```
Comment (randomJoe211) [inline docs_2.0/manual_EN/4.ngql-guide/6.functions/2.string.md:26]: Lacking article: return an empty string

File: docs_2.0/manual_EN/4.ngql-guide/6.functions/2.string.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Amber1990Zhang) [reply docs_2.0/manual_EN/4.ngql-guide/6.functions/2.string.md:12]: I think no NULL for Alpha

File: docs_2.0/manual_EN/4.ngql-guide/6.functions/2.string.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+string trim(string a) | Remove leading and trailing spaces |
+string ltrim(string a) | Remove leading spaces |
+string rtrim(string a) | Remove trailing spaces |
+string left(string a, int count) | Return the substring in [1, count], if length a is less than count, return a |
+string right(string a, int count) | Return the substring in [size - count + 1, size], if length a is less than count, return a |
+string lpad(string a, int size, string letters) | Left-pads a string with another string to a certain length|
+string rpad(string a, int size, string letters)| Reft-pads a string with another string to a certain length  |
+string substr(string a, int pos, int count) | Extract a substring from a string, starting at the specified position, extract the specified length characters |
+int hash(string a) | Encode the data into integer value |
+
```
[the remark this answers is not in the document]
Comment (Amber1990Zhang) [reply docs_2.0/manual_EN/4.ngql-guide/6.functions/2.string.md:22]: failed

**Review by randomJoe211 — APPROVED** [review]

**Review by whitewum — APPROVED** [review]

**Pull request merged**